src/sonic-yang-models/yang-models/sonic-system-port.yang

```diff
@@ -40,13 +40,8 @@ module sonic-system-port {
                 }
 
                 leaf ifname {
-                    type union {
-                        type leafref {
-                            path /port:sonic-port/port:PORT/port:PORT_LIST/port:name;
-                        }
-                        type string {
-                            pattern "Cpu0";
-                        }
+                    type string {
+                        length 1..128;
                     }
```

Comment:
Wouldn't it be better to add a reference to voq interface name?
```suggestion
			type leafref {
				path "/voq-inband-intf:sonic-voq-inband-interface/voq-inband-intf:VOQ_INBAND_INTERFACE/voq-inband-intf:VOQ_INBAND_INTERFACE_LIST/voq-inband-intf:name";
			}
                    }
```

you'll also need to add 
```
    import sonic-voq-inband-interface {
        prefix voq-inband-intf;
    }
```
at the import section of this file

Reply:
This problem is not related only to Inband port.If we have a scenario where one Linecard on the chassis has 36 ports and another linecard has 48 ports, then this leafref does not hold true.

Reply:
Why relax and remove the constrains?